src/trace/dma-trace.c

```diff
@@ -224,13 +224,18 @@ static int dma_trace_start(struct dma_trace_data *d)
 	config.dest_width = sizeof(uint32_t);
 	config.cyclic = 0;
 
-	err = dma_sg_alloc(&config.elem_array, SOF_MEM_ZONE_SYS,
-			   config.direction,
-			   elem_num, elem_size, elem_addr, 0);
+	err = dma_sg_array_alloc(&config.sg_array, 1, SOF_MEM_ZONE_SYS);
```

Comment:
dma_trace_start is called only once, at boot time, right? (code was like that before but still gonna ask to be sure)

Reply:
It's called on IPC

Reply:
And IPC only runs on master core. Alright then.

Reply:
And this IPC is only called once when boot is complete, right? Then this is fine.

Reply:
I'm afraid I'm not sure

Reply:
Kernel controls this flow and will only send this IPC once, then only those to enable and disable and thus this function will not be called again wasting SYS heap (free doesn't work with this one, that's why I was concerned)